hls-plugin-api/src/Ide/Plugin/Config.hs

```diff
@@ -79,7 +90,18 @@ parseConfig defValue = A.withObject "Config" $ \v -> do
         <*> (o .:? "checkProject" <|> v .:? "checkProject") .!= checkProject defValue
         <*> o .:? "formattingProvider"                      .!= formattingProvider defValue
         <*> o .:? "maxCompletions"                          .!= maxCompletions defValue
-        <*> o .:? "plugin"                                  .!= plugins defValue
+        <*> explicitParseFieldMaybe (parsePlugins $ plugins defValue) o "plugin" .!= plugins defValue
+
+parsePlugins :: Map.Map T.Text PluginConfig -> Value -> A.Parser (Map.Map T.Text PluginConfig)
```

Comment:
```suggestion
-- | Parse the 'PluginConfig'. Since we need to fall back to default values if we do not find one in the input, we need the map of plugin-provided defaults, as in 'defConfig'.
parsePlugins :: Map.Map T.Text PluginConfig -> Value -> A.Parser (Map.Map T.Text PluginConfig)
```